internal/pkg/controller/controller.go

```diff
@@ -1,25 +1,60 @@
 package controller
 
 import (
+	"log"
+	"time"
+
+	"github.com/stakater/Jamadaar/internal/pkg/actions"
 	"github.com/stakater/Jamadaar/internal/pkg/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	clientset "k8s.io/client-go/kubernetes"
 )
 
+const jamadaarDisableAnnotation = "jamadaar.stakater.com/persist"
+
+// Controller Jamadaar Controller to check for left over items
 type Controller struct {
 	clientset clientset.Interface
 	config    config.Config
+	Actions   []actions.Action
 }
 
-// NewController for initializing a Controller
+// NewController for initializing the Controller
 func NewController(clientset clientset.Interface, config config.Config) (*Controller, error) {
 	controller := &Controller{
 		clientset: clientset,
 		config:    config,
 	}
+	controller.Actions = actions.PopulateFromConfig(config.Actions)
 	return controller, nil
 }
 
-//Run function for controller which handles the queue
+//Run function for controller which handles the logic
 func (c *Controller) Run() {
-
+	for {
+		nameSpaceList, err := c.clientset.CoreV1().Namespaces().List(metav1.ListOptions{})
+		if err != nil {
+			log.Printf("Error getting namespaces: %v", err)
+			return
+		}
+		for _, namespace := range nameSpaceList.Items {
```

Comment:
@kahootali plz create a separate unit testable method which just handles this particular jamaadar task of the the cleanup of namespaces; and then write unit test for it

I mean pass it a list of namespaces so, that you can test it's core logic without running it inside k8s

this controller will have lots of other tasks of cleanup so, proper design of the solution is very important; maybe have a strategy interface and then each cleanup task should implement it?